### ISSUE TITLE
Open source MSTest source generator and Native AOT engine

### ISSUE THREAD

**Issue by Evangelink** [microsoft/testfx#2721]
## Summary

If you using or would like to use MSTest Native AOT support and the fact it is currently closed-source is problematic for you, please upvote this issue and/or add a comment.

**Comment by Nirmal4G** [comment]
@Evangelink How is it possible that you up-voted and down-voted at the same time!? 🧐 How does GitHub allow this? Usually it removes the up-vote if you down-vote and vice versa. Only for those two emojis. Somebody @Github broke this! 😂

**Comment by Evangelink** [comment]
The various emojis are not mutually exclusive. I have upvoted and downvoted to make it easy for people to upvote/downvote as they don't have to look it up.

**Comment by Nirmal4G** [comment]
No, I distinctly remember that only those two were infact, mutually exclusive. Once I accidentally selected the thumbs down emoji for the post (*in MSBuild repo when it was under @Microsoft*) I already up-voted, to mark with eyes and it removed the upvote. Then I selected the upvote again. It was the first and last time I did that. I thought it was really cool that they looked into these details like that!

I remember it was way before the New UI refresh, Dark mode and definitely before covid. So, something must have happened after.

**Comment by Nirmal4G** [comment]
Going a little off topic there... Back to the point in discussion, it's not very much as problematic than it would aid in customization, debugging and possibly contributing back to source.

So, yes... Open source all the things.

**Comment by Mpdreamz** [comment]
@Evangelink is there more background information to this vote? Is the default to open source things within testfx? Why was it released closed source to begin with?

**Comment by Evangelink** [comment]
> @Evangelink is there more background information to this vote? [...] Why was it released closed source to begin with?

Management/Company request. I think this should be OSS and I am hoping that getting enough upvotes would help me defend the case.

> Is the default to open source things within testfx? 

Yes. In the perfect world, I would have this repo renamed as `mstest` and would love to have another one like `dotnet/testing` that would host the new testing platform (behind MSTest runner) as it's now used by most of the .NET test frameworks.

**Comment by Youssef1313** [comment]
MSTest.SourceGeneration and MSTest.Engine are now part of this repo (PR https://github.com/microsoft/testfx/pull/4966)

**Comment by Mpdreamz** [comment]
Great to hear!
